File: modules/webrtc-core/src/main/java/com/shterneregen/securelan/webrtc/service/RtcSessionService.java
package com.shterneregen.securelan.webrtc.service;

import com.shterneregen.securelan.common.model.rtc.RtcSignalEnvelope;
import com.shterneregen.securelan.webrtc.runtime.RtcRuntimeStatus;

import java.util.Optional;

public interface RtcSessionService extends AutoCloseable {
    RtcRuntimeStatus runtimeStatus();
    Optional<RtcSessionSnapshot> currentSession();
    RtcSessionSnapshot startSession(RtcSessionRequest request);
    void acceptInboundSignal(String localPeer, RtcSignalEnvelope signal);
    void sendDataMessage(String payload);
    void closeCurrentSession();

    @Override
    default void close() {
    }
}

File: modules/webrtc-core/src/main/kotlin/com/shterneregen/securelan/webrtc/service/RtcSessionService.kt
package com.shterneregen.securelan.webrtc.service

import com.shterneregen.securelan.common.model.rtc.RtcSignalEnvelope
import com.shterneregen.securelan.webrtc.runtime.RtcRuntimeStatus
import java.util.Optional

interface RtcSessionService : AutoCloseable {
    fun runtimeStatus(): RtcRuntimeStatus
    fun currentSession(): Optional<RtcSessionSnapshot>
    fun startSession(request: RtcSessionRequest): RtcSessionSnapshot
    fun acceptInboundSignal(localPeer: String?, signal: RtcSignalEnvelope)
    fun sendDataMessage(payload: String?)
    fun closeCurrentSession()

    override fun close() {
    }
}
